sdk/storage/Azure.Storage.Files/src/Models/NtfsFileAttributes.cs

```diff
@@ -60,202 +52,151 @@ public class NtfsFileAttributes
         /// </summary>
-        public bool NotContentIndexed { get; set; }
+        NotContentIndexed = 256,
 
         /// <summary>
         /// The file or directory is excluded from the data integrity scan. When this value
         /// is applied to a directory, by default, all new files and subdirectories within
         /// that directory are excluded from data integrity.
         /// </summary>
-        public bool NoScrubData { get; set; }
-
-        /// <summary>
-        /// Checks if two FileNtfsAttributes are equal.
-        /// </summary>
-        /// <param name="other">The other instance to compare to.</param>
-        /// <returns></returns>
-        public override bool Equals(object other)
-            => other is NtfsFileAttributes attributes && Equals(attributes);
-
-        /// <summary>
-        /// Checks if two FileNtfsAttributes are equal to each other.
-        /// </summary>
-        /// <param name="other">TThe other instance to compare to.</param>
-        /// <returns></returns>
-        public bool Equals(NtfsFileAttributes other)
-            => None == other.None
-               && ReadOnly == other.ReadOnly
-               && Hidden == other.Hidden
-               && System == other.System
-               && Normal == other.Normal
-               && Directory == other.Directory
-               && Archive == other.Archive
-               && Temporary == other.Temporary
-               && Offline == other.Offline
-               && NotContentIndexed == other.NotContentIndexed
-               && NoScrubData == other.NoScrubData;
-
-        /// <summary>
-        /// Get a hash code for the FileNtfsAttributes.
-        /// </summary>
-        /// <returns>Hash code for the FileNtfsAttributes.</returns>
-        [EditorBrowsable(EditorBrowsableState.Never)]
-        public override int GetHashCode() =>
-            (None ? 0b00000000001 : 0) +
-            (ReadOnly ? 0b00000000010 : 0) +
-            (Hidden ? 0b00000000100 : 0) +
-            (System ? 0b00000001000 : 0) +
-            (Normal ? 0b00000010000 : 0) +
-            (Directory ? 0b00000100000 : 0) +
-            (Archive ? 0b00001000000 : 0) +
-            (Temporary ? 0b00010000000 : 0) +
-            (Offline ? 0b00100000000 : 0) +
-            (NotContentIndexed ? 0b01000000000 : 0) +
-            (NoScrubData ? 0b10000000000 : 0);
+        NoScrubData = 512
+    }
+}
 
+namespace Azure.Storage.Files
+{
+    /// <summary>
+    /// File enum extensions.
+    /// </summary>
+    internal static partial class FileExtensions
+    {
         /// <summary>
         /// ToString
         /// </summary>
         /// <returns>string</returns>
-        public override string ToString()
+        public static string ToAttributesString(this NtfsFileAttributes attributes)
         {
             var stringBuilder = new StringBuilder();
 
-            if (None)
```

Comment:
You need to keep None here.  null means unspecified, None means the attributes are empty.

Reply:
But none is the default value we pass to the service if the enum is null. Why would someone need to specify None explicitly?

Reply:
Is that what we do?  The way I understood it, when it's null we shouldn't pass anything (which leaves the value alone) and when it's `None` we should pass that (which will clear any value that's there).

Reply:
For the SetHttpHeaders it defaults to Preserve, for Create it defaults to None.
I think the Normal enum value is essentially the same as the None value. Should we delete Normal and update description of None? I don't see where Normal is actually referenced in the docs - https://docs.microsoft.com/en-us/rest/api/storageservices/create-directory

Reply:
@seanmcc-msft - can you weigh in here?  I remember we discussed [this awhile back](https://github.com/Azure/azure-sdk-for-net/pull/7361#discussion_r316425454) and maybe I'm getting it wrong.

Reply:
I agree with Josh, null is supposed to mean leave the current NTFS attributes.  We don't need None in the enum.

Reply:
I replaced Normal with None.